ludwig/data/preprocessing.py

```diff
@@ -1647,32 +1648,45 @@ def preprocess_for_training(
         with backend.storage.cache.use_credentials() if cached else contextlib.nullcontext():
             logger.debug("create training dataset")
             training_dataset = backend.dataset_manager.create(training_set, config, training_set_metadata)
-            if not len(training_set):
+            training_set_size = len(training_dataset)
+            if training_set_size == 0:
                 raise ValueError("Training data is empty following preprocessing.")
+            elif training_set_size < MIN_DATASET_SPLIT_ROWS:
+                raise ValueError(f"Training dataset has only {training_set_size} rows following preprocessing.")
```

Comment:
Should we add that it needs a minimum of X rows as a part of the error message?

Reply:
Good call, thanks!